RoddyCore/src/de/dkfz/roddy/client/cliclient/clioutput/CommandLineParameterParser.groovy

```diff
@@ -0,0 +1,303 @@
+        this.value = value
+    }
+
+}
+
+/** CValue is a value object and not modifiable. Therefore there are equals and hashCode methods. */
+@CompileStatic
+class CValue {
+    private final String name
+    private final String value
+    private final String type
+
+    CValue(String name, String value = null, String type = null) {
+        assert(null != name)
+        this.name = name
+        this.value = value
+        this.type = type
+    }
+
+    String getName() {
+        this.name
+    }
+
+
+    String getValue() {
+        this.value
+    }
+
+    String getType() {
+        this.type
+    }
+
+    String toString() {
+        "CValue('$name','$value','$type')"
+    }
+
+    @Override
+    int hashCode() {
+        Objects.hash(name, value, type)
+    }
+
+    @Override
+    boolean equals(Object obj) {
+        if(obj == null)
+            return false
+        if(!(obj instanceof CValue))
+            return false
+
+        CValue other = obj as CValue
+        name == other.name &&
+                value == other.value &&
+                type == other.type
+    }
+
+}
+
+@CompileStatic
+class CValueParameter extends ParameterWithValue {
+
+    Map<String, CValue> cvalues
+
+
+    private static String cvalueMapToValue(Map<String, CValue> cvalues) {
+        cvalues.collect { k, v ->
+            def res = v.name + ':' + v.value
+            if (v.type != null) {
+                res += ':' + v.type
+            }
+            res
+        }.join(',')
+    }
+
+    CValueParameter(Map<String, CValue> cvalues) {
+        super('cvalues', cvalueMapToValue(cvalues))
+        this.cvalues = cvalues
+    }
+
+}
+
+
+/** A parser for commandline values. The assumptions are
+ *
+ * * Parameters are provide in the form '--parameter=value' to the parser.
+ * * On the Shell command-line the parameters are first evaluated according to the shell's evaluation rules,
+ *   e.g. concerning quoting, escaping, dollar. E.g. '--parameter="varA:valA"' becomes '--parameter=varA:valA'
+ * * Generally, empty values are allowed and accepted by the parser. We leave it to the external code to
+ *   decide about the semantics. Also the commonly used '--' parameter is allowed.
+ *
+ * Implementation:
+ * * flatten internal values to strings where it makes sense (i.e. where the values are of use as unit) as early as possible (i.e. deep in the tree)
+ * * start transformations of input (map(), flatten(), etc.) in a new line
+  */
+@CompileStatic
+class CommandLineParameterParser {
+
+    /**
+     * Arbitrary parameters: .*
+     *
+     * "" is for the special case of roddy.sh ... '' ... . The parser should not break on this input. Its semantics
+     * (if any)  is decided elsewhere.
+      */
+    static Parser arbitraryParameterExpr = any().star().end().         // .*
+            flatten().
+            map { String m ->
+                new ArbitraryParameter(m)
+            }
+
+    static Parser parameterPrefix = of('-' as Character).times(2)
```

Comment:
This should be possible. But be aware, that the current parser only discriminates between the four types arbitrary parameters, parameters with values, parameters without values, and cvalues-parameters. No detailed parameters are considered. The main aim was to deal with cvalues.
At some point we should rather consider using an existing command-line parser system and reuse the cvalues-code here to check the syntax in the cvalues value. I think this would be the more scalable solution.